proto/spire/api/types/federationrelationship.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+package spire.api.types;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/types";
+
+import "spire/api/types/bundle.proto";
+
+message FederationRelationship {
+    // The trust domain name (e.g., "example.org") to federate with. 
+    string trust_domain = 1;
+
+    // URL of the SPIFFE bundle endpoint that provides the trust
+    // bundle to federate with. Must use the HTTPS protocol.
+    string bundle_endpoint_url = 2;
+
+    // Required. The endpoint profile type.
+    oneof bundle_endpoint_profile {
+        // Use Web PKI endpoint profile. 
+        HTTPSWebProfile https_web = 3;
+
+        // Use SPIFFE Authentication endpoint profile.
+        HTTPSSPIFFEProfile https_spiffe = 4;
+    }
+}
+
+message HTTPSSPIFFEProfile {
+    // Specifies the expected SPIFFE ID of the SPIFFE bundle endpoint server.
+    string endpoint_spiffe_id = 1;
+
+    // The bundle of the trust domain that will be used when performing requests
+    // to the SPIFFE bundle endpoint server. It will be persisted for future requests.
+    // If there is an existing stored bundle, this bundle will be used instead and the
+    // existing bundle will be updated. If not specified, a previously
```

Comment:
i think this partial suggestion is more accurate? Or will we use the specified bundle only on the first invocation and not write it to the bundle table prior to that?

```suggestion
    // If there is an existing stored bundle, this bundle will overwrite it. If
    // not specified, a previously
```

Reply:
Thanks for the suggestion, I meant to overwrite it :)